### PR TITLE
Fixes a race condition in the contract deployer when calling `Market.getTypeName`

### PR DESCRIPTION
**Pull request by adrake33**

### CODE REVIEW

File: source/libraries/ContractDeployer.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 #!/usr/bin/env node
 
+// TODO: Add type mappings for array items
```
Comment (MicahZoltu) [inline source/libraries/ContractDeployer.ts]: Is this comment related to binascii or something else?

Comment (adrake33) [reply source/libraries/ContractDeployer.ts]: No, this is in reference to the conversation that you, Jack, and I were having yesterday regarding loops where `hasOwnProperty` is not being checked.  I just put it here at the top of the file to serve as a more visible reminder.
